test/integration-tests/UniversalRouter.test.ts

```diff
@@ -1,270 +1,350 @@
+    let to = alice.address;
+    let deadline = Math.round((new Date()).getTime() / 1000) + 86400;
+    await (await daiContract.connect(alice).mint(alice.address, amountADesired)).wait();
+    await (await wethContract.connect(alice).deposit({ value: amountBDesired })).wait();
+
+    await (await daiContract.connect(alice).approve(uniswapV2Router.address, MAX_UINT)).wait();
+    await (await wethContract.connect(alice).approve(uniswapV2Router.address, MAX_UINT)).wait();
+
+    await (await uniswapV2Router.connect(alice).addLiquidity(
+      tokenA,
+      tokenB,
+      amountADesired,
+      amountBDesired,
+      amountAMin,
+      amountBMin,
+      to,
+      deadline
+    )).wait();
+
+    let reserves = await uniswapV2Router.getReserves(tokenA, tokenB);
+
+    expect(reserves[0]).to.be.eq(amountADesired);
+    expect(reserves[1]).to.be.eq(amountBDesired);
+
+    pairAddress = await uniswapV2Factory.getPair(daiContract.address, wethContract.address);
+    // console.log("pairAddress: " + pairAddress);
+    pair_DAI_WETH = pairAddress;
+
+    mockERC721 = await deployer.deploy(MockERC721, [alice.address]) as Contract;
+
+    mockSeaport = await deployer.deploy(MockSeaport, [cryptoCovens.address, alice.address]) as Contract;
+
+    permit2 = (await deployPermit2()).connect(alice) as Permit2;
+    router = (await deployRouter(
+      permit2,
+      wethContract.address, //weth9
+      mockSeaport.address,  // seaport
+      mockERC721.address,  //nftxzap
+      ZERO_ADDRESS, // x2y2
+      ZERO_ADDRESS, // foundation
+      ZERO_ADDRESS, // sudoswap
+      ZERO_ADDRESS, // nft20zap
+      ZERO_ADDRESS, // cryptopunks
+      ZERO_ADDRESS, // looksRare 
+      alice.address,  // routerRewardsDistributor
+      mockLooksRareRewardsDistributor.address, //looksRareRewardsDistributor 
+      mockLooksRareToken.address, // looksRareTokenAddress
+      uniswapV2Factory.address, // v2FactoryAddress
+      ZERO_ADDRESS, // uniswapV3
+    )).connect(alice) as UniversalRouter;
+
+    await (await daiContract.connect(alice).mint(alice.address, expandTo18DecimalsBN(1_000_000_000))).wait();
+    await (await wethContract.connect(alice).deposit({ value: expandTo18DecimalsBN(1_000_000) })).wait();
   })
 
   describe('#execute', () => {
-    let planner: RoutePlanner
+    let planner: RoutePlanner;
 
     beforeEach(async () => {
-      planner = new RoutePlanner()
-      await daiContract.approve(permit2.address, MAX_UINT)
-      await permit2.approve(DAI.address, router.address, MAX_UINT160, DEADLINE)
+      planner = new RoutePlanner();
+
+
+      await daiContract.mint(router.address, 1000000);
+      await daiContract.mint(permit2.address, 1000000);
+      await daiContract.approve(permit2.address, MAX_UINT);
+      await wethContract.approve(permit2.address, MAX_UINT);
+      await permit2.approve(daiContract.address, router.address, MAX_UINT160, DEADLINE);
+      await permit2.approve(wethContract.address, router.address, MAX_UINT160, DEADLINE);
```

Comment:
Are these mints needed on the router and permit2 addresses? And why you also approved the weth token?